### PR TITLE
rework some of the padding code

### PR DESCRIPTION
**Pull request by paultag**
Has a few things I'd like to clean up

### CODE REVIEW

**Comment by paultag** [comment]
```ruby
irb(main):004:0> def fnord(x)
irb(main):005:1>    x << "foo"
irb(main):006:1>    x
irb(main):007:1> end
=> :fnord
irb(main):008:0> x = "bar"
=> "bar"
irb(main):009:0> fnord x
=> "barfoo"
irb(main):010:0> x
=> "barfoo"
irb(main):011:0> 
```

File: spec/soap-scum_spec.rb [hunk 1]
```diff
@@ -271,10 +271,22 @@
...
-        expect(@message_processor.remove_xmlenc_padding(4, "\xA0\xB0\x03\xCCabc\x01\x04\x04\x04")).to eq('abc')
-        expect(@message_processor.remove_xmlenc_padding(4, "\xA0\xB0\x03\xCCabcd\xA0\xB0\xC0\x04")).to eq('abcd')
+        expect(@message_processor.remove_xmlenc_padding(4, "abcda\xA0\xB0\x03\x04")).to eq('abcda')
+        expect(@message_processor.remove_xmlenc_padding(4, "abcdab\xA0\x02\x04\x04")).to eq('abcdab')
+        expect(@message_processor.remove_xmlenc_padding(4, "abcdabc\x01\x04\x04\x04")).to eq('abcdabc')
+        expect(@message_processor.remove_xmlenc_padding(4, "abcdabcd\xA0\xB0\xC0\x04")).to eq('abcdabcd')
+      end
+
+      it 'can round trip a string' do
+        test_string = <<-jabber
```
Comment (awong-dev) [inline spec/soap-scum_spec.rb]: Yank the string into a let variable?  This is ugly to read.

File: src/soap-scum.rb [hunk 2]
```diff
@@ -267,8 +267,9 @@ def decrypt_cipher_text(decipher, key, iv, cipher_text)
       decipher.padding = 0
 
       plain = decipher.update(cipher_text)
-      plain << decipher.final 
-
+      plain << decipher.final
+      iv = plain.byteslice(0, decipher.block_size)
```
Comment (awong-dev) [inline src/soap-scum.rb]: This looks like a dead-store into iv....

Comment (paultag) [inline src/soap-scum.rb]: I saw `iv` was on another object, I was considering putting that somewhere, but I think we already grab it. Should I just kill that line and let the next line handle rewriting plain? I'll add some comments if so

Comment (awong-dev) [inline src/soap-scum.rb]: I definitely wouldn't deadstore a variable. That just reads like a coding error.

I would add a comment.

Comment (paultag) [inline src/soap-scum.rb]: Great, sending a PR in soon

**Pull request merged**